src/js/other-websites/inpage_menu.js

```diff
@@ -51,226 +52,610 @@ async function isUserSignedIn() {
   return userApiToken.hasOwnProperty("apiToken");
 }
 
-async function inpageContentInit() {
-  const sendInPageEvent = (evtAction, evtLabel) => {
-    sendRelayEvent("In-page", evtAction, evtLabel);
-  };
+async function getServerStoragePref() {
+  const serverStoragePref = await browser.storage.local.get("server_storage");
+  const serverStoragePrefInStorage =
+    serverStoragePref.hasOwnProperty("server_storage");
 
-  const { relaySiteOrigin } = await browser.storage.local.get(
-    "relaySiteOrigin"
-  );
-
-  // Set custom fonts from the add-on
-  await setCustomFonts();
+  if (!serverStoragePrefInStorage) {
```

Comment:
This fix was a time saver. Previously, we were calling the API every single time to see if the user has opt'd in/out to syncing labels. Rather than make that _long-distance call_, we now just check the local storage variable and only call when necessary.

Reply:
The add-on already injects an event listener on the add-on-specific custom element, which the website can trigger. It's currently used in one place (when the user updates a label, but server-side storage is disabled), see [how the website triggers the event](https://github.com/mozilla/fx-private-relay/blob/071c04326c09ae9bb9097d084f7345e2760e5cba/frontend/src/hooks/localLabels.ts#L53-L57) which [the add-on then processes](https://github.com/mozilla/fx-private-relay-add-on/blob/3ec2e7d9f3bbb807e30c165f18cf39033104573e/src/js/relay.firefox.com/get_profile_data.js#L462-L476). We could do something similar when the user flips a pref in the settings page.